ffi/diplomat/src/properties_maps.rs

```diff
@@ -57,6 +58,72 @@ pub mod ffi {
             self.0.as_borrowed().get32(cp)
         }
 
+        /// Converts a general category to its corresponding mask value
+        ///
+        /// Nonexistant general categories will map to the empty mask
+        #[diplomat::rust_link(icu::properties::GeneralCategoryGroup, Struct)]
+        pub fn general_category_to_mask(gc: u8) -> u32 {
+            if let Ok(gc) = GeneralCategory::try_from(gc) {
+                let group: GeneralCategoryGroup = gc.into();
+                group.into()
+            } else {
+                0
+            }
+        }
+
+        /// Produces an iterator over ranges of code points that map to `value`
+        #[diplomat::rust_link(
+            icu::properties::maps::CodePointMapDataBorrowed::iter_ranges_for_value,
+            FnInStruct
+        )]
+        pub fn iter_ranges_for_value<'a>(&'a self, value: u8) -> Box<CodePointRangeIterator<'a>> {
+            Box::new(CodePointRangeIterator(Box::new(
+                self.0.as_borrowed().iter_ranges_for_value(value),
+            )))
+        }
+
+        /// Produces an iterator over ranges of code points that do not map to `value`
+        #[diplomat::rust_link(
+            icu::properties::maps::CodePointMapDataBorrowed::iter_ranges_for_value_complemented,
+            FnInStruct
+        )]
+        pub fn iter_ranges_for_value_complemented<'a>(
+            &'a self,
+            value: u8,
+        ) -> Box<CodePointRangeIterator<'a>> {
+            Box::new(CodePointRangeIterator(Box::new(
+                self.0
+                    .as_borrowed()
+                    .iter_ranges_for_value_complemented(value),
+            )))
+        }
+
+        /// Given a General Category Mask value (obtained via `general_category_to_mask()` or
+        /// by using `ICU4XGeneralCategoryNameToMaskMapper`, produce an iterator over ranges of code points
+        /// whose `General_Category` values are contained in the mask.
+        ///
+        /// Should only be used on maps obtained via `load_general_category()`, other maps will have unpredictable results
+        #[diplomat::rust_link(
+            icu::properties::maps::CodePointMapDataBorrowed::iter_ranges_for_group,
+            FnInStruct
+        )]
+        pub fn iter_ranges_for_general_category_mask<'a>(
```

Comment:
Thought/Suggestion: you could generalize this and call it `iter_ranges_for_mask`, and then it works for any property whose values are all 31 or lower.

Reply:
seems fair, i was trying to not overcommit to masks but that's fine